libs/model-assessment/src/lib/ModelAssessmentDashboard/Cohort/GenerateTimeSeriesCohorts.ts

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation.
+// Licensed under the MIT License.
+
+import {
+  ErrorCohort,
+  JointDataset,
+  IFilter,
+  FilterMethods,
+  Cohort,
+  CohortSource,
+  IDataset
+} from "@responsible-ai/core-ui";
+
+import { translatePreBuiltCohortFilterForDataset } from "./ProcessPreBuiltCohort";
+
+export function generateTimeSeriesCohorts(
+  defaultErrorCohort: ErrorCohort,
+  preBuiltErrorCohortList: ErrorCohort[],
+  dataset: IDataset,
+  jointDataset: JointDataset
+): ErrorCohort[] {
+  if (
+    dataset.feature_metadata?.time_series_id_features &&
+    dataset.feature_metadata?.time_series_id_features?.length > 0
```

Comment:
mega nit: second dataset.feature_metatdata.time_series_id_features?.length (doesn't need the first ?)